etna/transforms/embeddings/models/ts2vec.py

```diff
@@ -0,0 +1,288 @@
+        depth:
+            The number of hidden residual blocks in the encoder.
+        device:
+            The device used for training and inference.
+        lr:
+            The learning rate.
+        batch_size:
+            The batch size.
+        n_epochs:
+            The number of epochs. When this reaches, the training stops.
+        n_iters:
+            The number of iterations. When this reaches, the training stops. If both n_epochs and n_iters are not specified,
+            a default setting would be used that sets n_iters to 200 for a dataset with size <= 100000, 600 otherwise.
+        verbose:
+            Whether to print the training loss after each epoch.
+        max_train_length:
+            The maximum allowed sequence length for training. For sequence with a length greater than ``max_train_length``,
+            it would be cropped into some sequences, each of which has a length less than ``max_train_length``.
+        temporal_unit:
+            The minimum unit to perform temporal contrast. When training on a very long sequence,
+            this param helps to reduce the cost of time and memory.
+        Notes
+        -----
+        In case of long series to reduce memory consumption it is recommended to use max_train_length parameter or manually break the series into smaller subseries.
+        """
+        super().__init__(output_dims=output_dims)
+        self.input_dims = input_dims
+        self.output_dims = output_dims
+        self.hidden_dims = hidden_dims
+        self.depth = depth
+        self.max_train_length = max_train_length
+        self.temporal_unit = temporal_unit
+
+        self.device = device
+        self.lr = lr
+        self.batch_size = batch_size
+
+        # Train params
+        self.n_epochs = n_epochs
+        self.n_iters = n_iters
+        self.verbose = verbose
+
+        self.embedding_model = TS2Vec(
+            input_dims=self.input_dims,
+            output_dims=self.output_dims,
+            hidden_dims=self.hidden_dims,
+            depth=self.depth,
+            max_train_length=self.max_train_length,
+            temporal_unit=self.temporal_unit,
+            device=self.device,
+            lr=self.lr,
+            batch_size=self.batch_size,
+        )
+
+    def _prepare_data(self, df: pd.DataFrame) -> np.ndarray:
+        """Convert data to array with shapes (n_segments, n_timestamps, input_dims)."""
+        n_timestamps = len(df.index)
+        n_segments = df.columns.get_level_values("segment").nunique()
+        df = df.sort_index(axis=1)
+        x = df.values.reshape((n_timestamps, n_segments, self.input_dims)).transpose(1, 0, 2)
+        return x
+
+    def fit(self, df: pd.DataFrame) -> "TS2VecEmbeddingModel":
+        """Fit the embedding model."""
+        x = self._prepare_data(df=df)
+        self.embedding_model.fit(train_data=x, n_epochs=self.n_epochs, n_iters=self.n_iters, verbose=self.verbose)
+        return self
+
+    def encode_segment(
```

Comment:
i decided to pass parameters for encode here.
the advantage is that we can call encode methods several times with different params with one model